src/platform/platform_posix.c

```diff
@@ -776,8 +854,20 @@ CxPlatCurThreadID(
     void
     )
 {
+
+#if defined(CX_PLATFORM_LINUX)
+
     CXPLAT_STATIC_ASSERT(sizeof(pid_t) <= sizeof(uint32_t), "PID size exceeds the expected size");
     return syscall(__NR_gettid);
+
+#elif defined(CX_PLATFORM_DARWIN)
+
+    uint64_t tid;
+    pthread_threadid_np(NULL, &tid);
+    // XXX: check for failure
+    return (uint32_t)tid;
```

Comment:
This may be problematic when it wraps IMHO. Did you consider changing CxPlatCurThreadID() to return uint64_t? That would cover the Linux as well.

Reply:
I will check these out tomorrow morning. The one who originally did the mac port did this, so I'll double check everything tomorrow. I'll check the other comment, and any other ones found as well.

Reply:
maybe add same assert as above. So if it wraps it would be visible. I don't know how this is consumed and probably does not need to addressed right now.